### PR TITLE
Add 'alt' to bidix E and monodix pardef . Fixes #8 . Fixes #1

### LINKED ISSUE #1: apertium-dixtools issue with alt
I'm doing:
apertium-dixtools format -alignMonodix apertium-oci.oci.metadix new.metadix

The problem are the lines like:

>         <pardef n="S__cantar" alt="oci@aran">

which are converted to

>         <pardef n="S__cantar">

This is a bug. alt should be conserved

### LINKED ISSUE #8: apertium-dixtools fix mangles entries with alt
When running `apertium-dixtools fix` on a bidix containing entries with an `alt` attribute, the attribute is ignored and the resulting entries are mangled. Take this minimal file from a real example in the `apertium-oci-cat` bidix:

```
<?xml version="1.0" encoding="utf-8"?>
<dictionary>
  <alphabet></alphabet>
  <sdefs>
    <sdef n="adj" /> 
    <sdef n="f" /> 
    <sdef n="m" /> 
    <sdef n="mf" /> 
    <sdef n="num" /> 
    <sdef n="GD" /> 
  </sdefs>
  <section id="main" type="standard">

<e r="LR" a="mmonge" alt="oci"><p><l>cruciau<s n="adj"/></l><r>crucial<s n="adj"/></r></p></e>
<e a="mmonge" alt="oci@aran"><p><l>cruciau<s n="adj"/></l><r>crucial<s n="adj"/></r></p></e>
<e r="LR" alt="oci@aran"><p><l>crucial<s n="adj"/><s n="m"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="LR" alt="oci"><p><l>crucial<s n="adj"/><s n="m"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="LR" alt="oci@aran"><p><l>crucial<s n="adj"/><s n="f"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="LR" alt="oci"><p><l>crucial<s n="adj"/><s n="f"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="RL" alt="oci"><p><l>crucial<s n="adj"/><s n="GD"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>

  </section>
  <section id="web" type="standard">
<e>       <re>[a-z.]+@[a-z.]+</re><p><l><s n="num"/></l><r><s n="num"/></r></p></e>
<e>       <re>((www.)[a-z\-]+[.][a-z]+)</re><p><l><s n="num"/></l><r><s n="num"/></r></p></e>
  </section>
</dictionary>
```
The resulting file has a mix of incorrect entries that generate testvoc errors:

```
<?xml version="1.0" encoding="utf-8"?>

<dictionary>
  <alphabet></alphabet>
  <sdefs>
    <sdef n="adj" /> 
    <sdef n="f" /> 
    <sdef n="m" /> 
    <sdef n="mf" /> 
    <sdef n="num" /> 
    <sdef n="GD" /> 
  </sdefs>
  <section id="main" type="standard">

<e a="mmonge" alt="oci"><p><l>cruciau<s n="adj"/></l>  <r>crucial<s n="adj"/></r></p></e>
<e r="LR" alt="oci@aran"><p><l>crucial<s n="adj"/><s n="m"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="LR" alt="oci@aran"><p><l>crucial<s n="adj"/><s n="f"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>
<e r="RL" alt="oci"><p><l>crucial<s n="adj"/><s n="GD"/></l><r>crucial<s n="adj"/><s n="mf"/></r></p></e>

  </section>
  <section id="web" type="standard">
<e>       <re>[a-z.]+@[a-z.]+</re><p><l><s n="num"/></l><r><s n="num"/></r></p></e>
<e>       <re>((www.)[a-z\-]+[.][a-z]+)</re><p><l><s n="num"/></l><r><s n="num"/></r></p></e>
  </section>
</dictionary>
```

### PR DESCRIPTION
**Pull request by xavivars**
This MR adds `alt` to `E` and to `pardef`.

It also updates java version (I was not able to make it work with the previous one, openjdk21 cannot build java6 anymore). And to do that, I had to change the library used for XML parsing.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -115,6 +115,11 @@
 	</build>
 	
 	<dependencies>
+		<dependency>
```
Comment (xavivars) [inline pom.xml:118]: This dependency was implicit, now is explicit.

File: src/dictools/utils/KeepTrackOfLocationDOMParser.java [hunk 2]
```diff
@@ -5,13 +5,12 @@
...
-import com.sun.org.apache.xerces.internal.xni.NamespaceContext;
-import com.sun.org.apache.xerces.internal.xni.QName;
-import com.sun.org.apache.xerces.internal.xni.XMLAttributes;
-import com.sun.org.apache.xerces.internal.xni.XMLLocator;
+import org.apache.xerces.parsers.DOMParser;
+import org.apache.xerces.xni.Augmentations;
+import org.apache.xerces.xni.NamespaceContext;
+import org.apache.xerces.xni.QName;
+import org.apache.xerces.xni.XMLAttributes;
+import org.apache.xerces.xni.XMLLocator;
```
Comment (xavivars) [inline src/dictools/utils/KeepTrackOfLocationDOMParser.java:13]: This change stops using Sun internal classes and uses the explicitly defined xerces dependency instead.

File: pom.xml [hunk 3]
```diff
@@ -74,8 +74,8 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-compiler-plugin</artifactId>
 				<configuration>
-					<source>1.6</source>
-					<target>1.6</target>
+					<source>1.7</source>
```
Comment (TinoDidriksen) [inline pom.xml:77]: Both need to be bumped to 1.8 it seems.

Comment (xavivars) [reply pom.xml:77]: It's weird, they were already in different versions, but I guess that makes sense

Comment (xavivars) [inline pom.xml:77]:
```suggestion
					<source>1.8</source>
```

File: pom.xml [hunk 4]
```diff
@@ -74,8 +74,8 @@
...
+					<target>1.7</target>
```
Comment (xavivars) [inline pom.xml:78]:
```suggestion
					<target>1.8</target>
```

**Review by TinoDidriksen — APPROVED** [review]

**Pull request merged**